### PR TITLE
remove restore from migrate tests

### PR DESCRIPTION
**Pull request by brthor**
Also remove Microsoft.NuGet.targets

We'll need to re-revert my change in SDK for this to work. 

@eerhardt @livarcocc

### CODE REVIEW

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 1]
```diff
@@ -237,7 +237,7 @@ private MigratedBuildComparisonData BuildProjectJsonMigrateBuildMSBuild(string p
             File.Delete(Path.Combine(projectDirectory, "project.lock.json"));
 
             MigrateProject(projectDirectory);
-            Restore(projectDirectory);
```
Comment (eerhardt) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: can we remove the `Restore` method?

Comment (brthor) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: no it's used to build the project.json to compare outputs

Comment (brthor) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: but it's not used for migration

Comment (eerhardt) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: ah right - forgot about that.

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/SaveOutputProjectRule.cs [hunk 2]
```diff
@@ -17,6 +17,9 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
 
             CleanEmptyPropertyAndItemGroups(migrationRuleInputs.OutputMSBuildProject);
 
+            // Temporarily Add ProjectLockFile
+            migrationRuleInputs.CommonPropertyGroup.AddProperty("ProjectLockFile", "$(MSBuildProjectDirectory)/$(MSBuildProjectName).project.lock.json");
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/SaveOutputProjectRule.cs]: I have this in my PR. you will see when you rebase. I added a WorkaroundRules, to be very clear that transformations there should be temporary.

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 3]
```diff
@@ -297,6 +297,15 @@ private void Restore(string projectDirectory)
                 .Pass();
         }
 
+        private void Restore3(string projectDirectory)
+        {
+            new TestCommand("dotnet")
```
Comment (livarcocc) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: I added this there as well. Check the rebase. I also added an actual Restore3Command test command.

**Comment by brthor** [comment]
This got superseded by other things

**Pull request closed without merging**